src/test/neorados/test_neorados.cc

```diff
@@ -27,13 +52,486 @@ TEST_F(TestNeoRADOS, MakeWithLibRADOS) {
   bufferlist bl;
   op.read(0, 0, &bl);
 
-  // provide pool that doesn't exist -- just testing round-trip
+  // provide pool that doesn't exists -- just testing round-trip
+  ASSERT_THROW(
+    rados.execute({"dummy-obj"}, IOContext(std::numeric_limits<int64_t>::max()),
+                  std::move(op), &bl, ceph::async::use_blocked),
+    boost::system::system_error);
+}
+
+TEST_F(TestNeoRADOS, MakeWithCCT) {
+  ceph::async::io_context_pool p(1);
+  auto rados = RADOS::make_with_cct(cct.get(), p,
+      ceph::async::use_blocked);
+
+  ReadOp op;
+  bufferlist bl;
+  op.read(0, 0, &bl);
+
+  // provide pool that doesn't exists -- just testing round-trip
+  ASSERT_THROW(
+      rados.execute({"dummy-obj"}, IOContext(std::numeric_limits<int64_t>::max()),
+                  std::move(op), &bl, ceph::async::use_blocked),
+      boost::system::system_error);
+}
+
+TEST_F(TestNeoRADOS, MakeWithBuilder) {
+  GTEST_SKIP(); // the build command hangs with the "use_blocked" flag
```

Comment:
If you're going to do blocking tests you need an `io_context_pool` so it will have a thread to run in and wake up the blocker.